src/applications/oscilloscope.py

```diff
@@ -1,117 +1,247 @@
+        show_rolling_mean: bool, default True
+            Whether to display the rolling mean line in the figure.
+        show_reading_text: bool, default True
+             Whether to display current reading value as text.
+        fig : matplotlib.figure.Figure
+            Figure to plot the data in.
+        ax : matplotlib.axes.Axes
+            Axis to plot the data in.
+        """
+
+        # setting up the figure and main axis
+        if ax is None:
             fig, ax = plt.subplots()
             self.fig = fig
             self.ax = ax
         else:
             self.fig = fig
             self.ax = ax
+        self.fig.set_layout_engine('compressed')
 
+        # display settings
+        self.show_rolling_mean = show_rolling_mean
+        self.show_reading_text = show_reading_text
+
+        # setting up primary data plot
         self.ydata = collections.deque(np.zeros(width))
         self.line, = self.ax.plot(self.ydata)
-        self.ax.set_ylabel('counts / sec')
-        self.ax.ticklabel_format(style='sci',scilimits=(-3,4),axis='y')
+        self.line: plt.Line2D
+
+        # Setting up axis preferences
+        # TODO: Put all axis preferences to .mplstyle and load through terminal or gui.
+        self.ax.set_ylabel('Reading (counts / sec)')
+        self.ax.ticklabel_format(style='sci', scilimits=(-3, 4), axis='y')
+        self.ax.set_xlabel('Acquisition Number')
+
+        if self.show_rolling_mean:
+            self._add_rolling_mean(rolling_mean_window_size)
+        if self.show_reading_text:
+            self._add_reading_text()
+
+    def _add_rolling_mean(self, rolling_mean_window_size: int):
+        """ Adds rolling mean line to figure. """
+        self.rolling_mean_window_size = rolling_mean_window_size
+        self.half_window_size = int(np.ceil(self.rolling_mean_window_size / 2))
+        self.rolling_mean = self.get_rolling_mean()
+        self.rolling_mean_line, = self.ax.plot(self.rolling_mean)
+        self.rolling_mean_line: plt.Line2D
+
+    def _add_reading_text(self):
+        """Adds current reading value as text on figure."""
+        self.text_font_size = 30  # TODO: Add to .mplstyle file
+        fig_width = self.fig.get_size_inches()[0]
+        fig_height = self.fig.get_size_inches()[1]
+        if self.show_rolling_mean:
+            self.fig.set_size_inches(fig_width, fig_height * 1.3)
+            displayed_values = 'NaN\nNaN\nNaN (NaN)'
+            displayed_labels = 'Cur. Val.\nMean\nSTD'
+        else:
+            self.fig.set_size_inches(fig_width, fig_height * 1.1)
+            displayed_values = 'NaN'
+            displayed_labels = 'Cur. Val.'
+        self.current_value_text: plt.Text = self.ax.text(
+            1, 1.05, displayed_values, fontsize=self.text_font_size, transform=self.ax.transAxes, ha='right')
+        self.ax.text(
+            0, 1.05, displayed_labels, fontsize=self.text_font_size, transform=self.ax.transAxes, ha='left')
```

Comment:
These text fields are huge.  Would it be too much to reduce this?   Also, can we make this scientific notation? And can we round the accuracy of the displayed value to the same order of magnitude of the STD?  Finally, the STD line has two values -- one value in parenthesis. What do these mean?

Reply:
- I wanted to make them this big because in our lab we have to look at the counts from all the way across the room, making it difficult to align otherwise. We can add a parameter to change that in the initialization arguments. 
- I think scientific notation might work. I will give it a try! 
- I like the rounding to STD magnitude. I will work on it.
- The first value is the measured STD. The second is the expected STD if the read rate was a poison distribution (which is usually the case for our measurements). That way you can tell how far off the measured STD is from the "ideal" source STD.